src/main/kotlin/com/bantolomeus/service/BookService.kt

```diff
@@ -16,84 +16,87 @@ class BookService(private val bookRepository: BookRepository,
 
     fun createBook(bookDTO: BookDTO): BookDTO {
         val savedBookDTO = bookRepository.saveBookIfItNotExists(bookDTO)
-        if (savedBookDTO.currentPage > 0L) {
+        if (savedBookDTO.currentPage > 0) {
             saveBookUpdate(savedBookDTO)
         }
         return bookDTO
     }
 
     fun updateBook(bookUpdate: BookUpdateInputDTO, bookName: String): BookUpdatesFileDTO {
-        var response = BookUpdatesFileDTO()
-        var books = bookRepository.getBooks().books.filter { it.name != bookName }
-        var oldPage = Long.MAX_VALUE
-        val foundBook = bookRepository.getBookByName(bookName)
+        val foundBook = bookRepository.getBookByName(bookName) ?: return BookUpdatesFileDTO()
```

Comment:
but what I will do, is rename `foundBook` to `book` because there is no need to add this `found` meaning to the variable. There's only one book variable. Before my changes there were two and a differentiation was needed

Reply:
Both points are ok. I do not know why I cam up with ```updateExistingBook```. You are right here.